airflow/models/taskmixin.py

```diff
@@ -88,3 +99,167 @@ def __init_subclass__(cls) -> None:
             stacklevel=2,
         )
         return super().__init_subclass__()
+
+
+class DAGNode(DependencyMixin):
+    """
+    A base class for a node in the graph of a workflow -- an Operator or a Task Group, either mapped or
+    unmapped.
+    """
+
+    dag: Optional["DAG"] = None
+
+    @property
+    @abstractmethod
```

Comment:
I think this is useless because `DAGNode` (and its parent `DependencyMixin`) does not use `ABCMeta`?

Reply:
Fair point -- think I should remove the decorator or change the meta class?

Reply:
I'd vote for adding ABCMeta. The DAGNode becomes an important class for us, and maybe some day we'll add more nodes, who knows.

Reply:
Made a real ABC.